### PR TITLE
Webhook Improvements

### PR DESCRIPTION
**Pull request by ghostzero**
Hey,

I've made some changes to the source code, we should discuss if it's so ok.

**CHANGELOG:**

* Added Game & Extension Analytics Topic
* I added $token to getWebhookSubscriptions() because it's required by this endpoint. @romanzipp Q: Should the token be passed as a parameter or rather with `withToken`? I don't know the best practice.
* The urlencode() is now implemented directly in `executeQuery`.
* I have updated the documentation because PhpStorm has shown me some bugs.

### CODE REVIEW

**Comment by romanzipp** [comment]
Concerning token usage with the `getWebhookSubscriptions` method:
You should add an optional token parameter just like `getAuthedUser(string $token = null)` if a method heavily rests on user authentication. This parameter should always be optional since you can set the token on the Twitch wrapper instance.

File: src/Twitch.php [hunk 1]
```diff
@@ -240,24 +243,24 @@ public function query(string $method = 'GET', string $path = '', array $paramete
...
-            $request = new Request($method, $uri, $headers, $jsonBody);
+        $request = new Request($method, $uri, $headers, $jsonBody);
 
+        try {
             $response = $this->client->send($request);
 
             $result = new Result($response, null, $paginator, $this->legacy ? true : false);
         } catch (RequestException $exception) {
             $result = new Result($exception->getResponse(), $exception, $paginator);
-        } catch (ClientException $exception) {
```
[the remark this answers is not in the document]
Comment (GhostZero) [reply src/Twitch.php]: 'RequestException', superclass of exception class 'ClientException', has already been caught

File: src/Traits/WebhooksTrait.php [hunk 2]
```diff
@@ -7,12 +7,35 @@
...
+     *     request to confirm that you requested the subscription. To confirm,
+     *     you must respond to the request with the challenge token provided in
+     *     the query parameters and an HTTP success (2xx) response code.
+     *
+     * Subscription denied:
+     *     In this case, the request payload explains why the subscription
+     *     could not be created. For example, you may not be authorized to access
+     *     the resource you requested or you may have exceeded the maximum number of subscriptions.
+     *
+     * @param string $callback URL where notifications will be delivered.
```
[the remark this answers is not in the document]
Comment (GhostZero) [reply src/Traits/WebhooksTrait.php]: Fixed :+1:

**Comment by GhostZero** [comment]
Hey, i removed the WIP flag. I think this PR will be a BC because of urlencode.

**Pull request closed without merging**